### PR TITLE
Add support for Consonant code typography

### PR DESCRIPTION
**Pull request by hparra**
Load font and styling to support [Consonant's code typography](https://consonant.adobe.com/1975e5ba1/v/32165/p/216c7e-typography/t/244b88) when document contains any inlined or blocked monospaced text.

Screenshot 2023-08-30 at 19 17 02

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.live/drafts/hgpa/test/code/code?martech=off
- After: https://feat-code-noblock--milo--hparra.hlx.live/drafts/hgpa/test/code/code?martech=off

### CODE REVIEW

**Comment by hparra** [comment]
I also made a code block #1176 that has most of this code, but the Consonant styling shouldn't depend on whether the code is in a block or not.

File: libs/styles/variables.css [hunk 1]
```diff
@@ -30,6 +30,7 @@
 
   /* Fonts */
   --body-font-family: 'Adobe Clean', adobe-clean, 'Trebuchet MS', sans-serif;
+  --code-font-family: 'Source Code Pro', source-code-pro, monospace;
   --fixed-font-family: 'Roboto Mono', menlo, consolas, 'Liberation Mono', monospace;
```
Comment (hparra) [inline libs/styles/variables.css:34]: I checked all repos under org and nothing is using `--fixed-font-family`. Delete?

File: libs/styles/variables.css [hunk 2]
```diff
@@ -97,6 +98,15 @@
...
   --type-detail-s-lh: 12px;
 
+  /* Code */
+  --type-code-all-weight: 400;
+  --type-code-l-size: 18px;
+  --type-code-l-lh: 27px;
+  --type-code-m-size: 16px;
+  --type-code-m-lh: 24px;
+  --type-code-s-size: 14px;
+  --type-code-s-lh: 21px;
```
Comment (overmyheadandbody) [inline libs/styles/variables.css:108]: All line heights are actually `1.5` when compared to the fonts. Do we even need them defined explicitly?

File: libs/utils/utils.js [hunk 3]
```diff
@@ -778,8 +778,12 @@ async function loadPostLCP(config) {
...
   }
   loadTemplate();
-  const { default: loadFonts } = await import('./fonts.js');
-  loadFonts(config.locale, loadStyle);
+  import('./fonts.js').then((module) => {
+    module.default(config.locale, loadStyle);
+    if (document.querySelector('code')) {
+      module.loadTypeKitFont('vih2anh.css', loadStyle);
+    }
+  });
```
Comment (overmyheadandbody) [inline libs/utils/utils.js:786]: Maybe a unit test would be valuable here to check that requests are made for the respective TypeKit fonts

File: libs/styles/variables.css (hunk 1 above)
Comment (overmyheadandbody) [reply libs/styles/variables.css:34]: HelpX seems to use it in their [styles](https://github.com/adobecom/helpx/blob/37b7eb6956405970321e53657b140e32c6784a1a/styles/styles.css#L108), but they also define it [themselves](https://github.com/adobecom/helpx/blob/37b7eb6956405970321e53657b140e32c6784a1a/styles/styles.css#L24), so this does seem redundant to me.

File: libs/styles/typography.css [hunk 4]
```diff
@@ -129,6 +132,27 @@
...
+
+.code-s {
+  font-size: var(--type-code-s-size);
+  line-height: var(--type-code-s-lh);
+}
+
+.code-l,
+.code-m,
+.code-s {
+  font-weight: var(--type-code-all-weight);
```
Comment (overmyheadandbody) [inline libs/styles/typography.css:153]: `line-height: 1.5` could go here directly.

**Review by overmyheadandbody — APPROVED** [review]

File: libs/styles/variables.css (hunk 2 above)
Comment (hparra) [reply libs/styles/variables.css:108]: I did not notice that. I’m following the conventions of the surrounding typography, which do not follow a consistent scale. Should we be explicit like Consonant specs or are we free to optimize? I’m ok with either way.

**Comment by hparra** [comment]
TODO: Rebase after https://github.com/adobecom/milo/pull/1194

**Comment by auniverseaway** [comment]
What shipping page needs this? I'm really struggling with all the additions we keep piling on to utils.js.

**Comment by hparra** [comment]
@auniverseaway It's part of Consonant. It should already be there whenever a page uses monospaced text.

**Review by auniverseaway — CHANGES_REQUESTED** [review]
1. We need a JIRA.
2. I don't think we should ship this until there's an actual need. I dislike shipping features that no one is actually asking for or is not tied to a page going live. "It's in Consonant" in my mind is not good enough... there's a lot of stuff in Consonant that doesn't exist in Milo.
3. If we are going to ship, I would not make utils more complicated, I would move the complexity into fonts.js.

**Comment by hparra** [comment]
Thanks for your review @overmyheadandbody. Closing.

**Pull request closed without merging**